ml-dsa/src/param.rs

```diff
@@ -31,20 +31,21 @@ use crate::encode::{
 use crate::util::{B32, B64};
 
 /// Some useful compile-time constants
-pub type SpecQ = Sum<Diff<Shleft<U1, U23>, Shleft<U1, U13>>, U1>;
-pub type SpecD = U13;
-pub type QMinus1 = Diff<SpecQ, U1>;
-pub type BitlenQMinusD = Diff<Length<SpecQ>, SpecD>;
-pub type Pow2DMinus1 = Shleft<U1, Diff<SpecD, U1>>;
-pub type Pow2DMinus1Minus1 = Diff<Pow2DMinus1, U1>;
+pub(crate) type SpecQ = Sum<Diff<Shleft<U1, U23>, Shleft<U1, U13>>, U1>;
+pub(crate) type SpecD = U13;
+pub(crate) type QMinus1 = Diff<SpecQ, U1>;
+pub(crate) type BitlenQMinusD = Diff<Length<SpecQ>, SpecD>;
+pub(crate) type Pow2DMinus1 = Shleft<U1, Diff<SpecD, U1>>;
+pub(crate) type Pow2DMinus1Minus1 = Diff<Pow2DMinus1, U1>;
 
 /// An integer that describes a bit length to be used in sampling
+#[expect(unreachable_pub)]
```

Comment:
There's a couple of instances of unreachable pub traits which need to remain pub due to trait constraints on associated types in pub traits. I've silenced the warning without making these reachable in these cases.